pallas-network/src/miniprotocols/localstate/queries_v16/mod.rs

```diff
@@ -491,16 +491,41 @@ pub enum Vote {
     Abstain,
 }
 
-pub type RewardAccount = Bytes;
+#[derive(Debug,  Clone, Eq, PartialEq, PartialOrd, Ord)]
+pub struct FieldedRewardAccount{
```

Comment:
is the `FieldedRewardAccount` struct defined as part of some spec or is it a quality-of-life improvement for dealing with credentials?

If its the latter, I would prefer to have this in a higher layer of the stack and not as part of the codec of the miniprotocol.

Reply:
@scarmuega It's there to be able to produce the same error message with Haskell codebase. Since Rust doesn't have type specialization, old-good RewardAccount (which is a Byte) can't be  used for HaskellDisplay to distinguish. 

So not sure if this is a quality-of-life improvement or not. If do you think so, I could refactor it. Just tell me what layer it should be in that case. Thanks.